core/src/main/scala/org/apache/spark/status/KVUtils.scala

```diff
@@ -100,6 +100,33 @@ private[spark] object KVUtils extends Logging {
     }
   }
 
+  /** Counts the number of elements in the KVStoreView which satisfy a predicate. */
+  def count[T](view: KVStoreView[T])(countFunc: T => Boolean): Int = {
```

Comment:
@srowen Do you mind if I split the newly introduced functions and related refactoring work into another pr? It seems that the change of this part does not increase the complexity of the code.

Reply:
I think that's OK

Reply:
OK ~

Reply:
@srowen I separate the code related to these 4 new functions into a new pr: https://github.com/apache/spark/pull/36237/files